canvas/biplot.py

```diff
@@ -0,0 +1,167 @@
+                             'not %r.' % type(sample_color_dict).__name__)
+
+    # if feature_color_dict is None.
+    # Specify a default palette (default colors).
+    # if (feature_color_dict == None):
+    if feature_color_dict is not None:
+        if not isinstance(feature_color_dict, dict):
+            raise ValueError('`feature_color_dict` must be a `dictionary`, '
+                             'not %r.' % type(feature_color_dict).__name__)
+
+    # check to see if pandas data frames are out of order,
+    # do sort here and re-check if data frames are different lengths?
+    if sample_df_metadata is not None:
+        if (sample_df.index != sample_df_metadata.index).any():
+            sample_df = sample_df.sort_index(axis=0)
+            sample_df_metadata = sample_df_metadata.sort_index(axis=0)
+
+    fig = plt.figure(figsize=(15, 15))
+    ax = fig.add_subplot(111)
+
+    if sample_df_metadata is None:
+        ax.plot(np.ravel(sample_df.ix[:, 0]), np.ravel(sample_df.ix[:, 1]),
+                marker='o', linestyle='', ms=8)
+    else:
+        for name, group in sample_df.groupby(sample_df_metadata[
+                                             samp_col_category]):
+            if sample_color_dict is not None:
+                ax.plot(np.ravel(group.ix[:, 0]), np.ravel(group.ix[:, 1]),
+                        marker='o', linestyle='', ms=8,
+                        color=sample_color_dict[name],
+                        label=name)
+            else:
+                ax.plot(np.ravel(group.ix[:, 0]), np.ravel(group.ix[:, 1]),
+                        marker='o', linestyle='', ms=8,
+                        label=name)
+    ax2 = ax.twinx()
+    if sample_df_metadata is not None:
+        ax.legend(title=sample_color_category, loc=2, numpoints=1)
+    ax2.set_ylim(ax.get_ylim())
+
+    recs = []
+    feature = []
+    otu_feature_color = []
+    otu_feature_append = []
+
+    # choose percent of arrows to show, feature,
+    # which arrows and labeling of arrows?
+    # feature_df =
+    # feature_df[feature_df['radius'] >= max(feature_df['radius'])*.001]
+
+    if feature_df_metadata is None:
+        for index, row in feat_df.iterrows():
+            ax2.arrow(0, 0, np.asscalar(row[0:1]), np.asscalar(row[1:2]),
+                      width=0.02, head_width=0.05, alpha=0.7)
+    else:
+        for name, group in
+        feature_df.groupby(feature_df_metadata[feature_color_category]):
+            if feature_color_dict is not None:
+                if len(name.split(';')) > 1:
+                    otu_feature = name.split(';')[1].replace(' ', '')
+                    otu_feature_color.append(feature_color_dict[otu_feature])
+                    otu_feature_append.append(otu_feature)
+                    ax2.arrow(0, 0, np.asscalar(group.ix[:, 0]),
+                              np.asscalar(group.ix[:, 1]),
+                              width=0.02, head_width=0.05, alpha=0.5,
```

Comment:
We should be able to pass in `width`, `head_width`, and `alpha` all as parameters.

If this is becoming overwhelming, we can exploit using the [`**kwargs`](http://stackoverflow.com/questions/1769403/understanding-kwargs-in-python)